pkg/cfn/builder/iam.go

```diff
@@ -439,6 +440,14 @@ func (rs *IAMServiceAccountResourceSet) AddAllResources() error {
 	return nil
 }
 
+func truncateResourceName(name string) string {
```

Comment:
If CloudFormation will prepend a unique suffix, it will truncate it automatically also. We shouldn't need this function.

Reply:
It does not prepend a unique suffix. The field needs to be set as mentioned below.